### ISSUE TITLE
wp-cron / loopback requests don't work (Curl error 7)

### ISSUE THREAD

**Issue by MinKaizen** [docker-library/wordpress#611]
## Description
wp-cron and "loopback requests" don't work when using this wordpress docker image.
docker-compose version: `3`
wordpress image version: `latest` (5.7.2 as of writing)
docker-compose file: See below. Courtesy of BradTraversy

## Error Message
In `Tools > Site Health` there are 2 errors:
```
The REST API encountered an error
The REST API request failed due to an error
Error: cURL error 7: Failed to connect to localhost port 8000: Connection refused (http_request_failed)
```
and also:
```
Your site could not complete a loopback request
The loopback request to your site failed, this means features relying on them are not currently working as expected.
Error: cURL error 7: Failed to connect to localhost port 8000: Connection refused (http_request_failed)
```

If you install the Plugin called `WP Crontrol` you will also see this error in `Tools > Cron Events`:
```
There was a problem spawning a call to the WP-Cron system on your site. This means WP-Cron events on your site may not work. The problem was:
cURL error 7: Failed to connect to localhost port 8000: Connection refused
```

## Steps to Reproduce

1. `mkdir wp-cron-test && cd wp-cron-test`
2. (Create the docker compose file below)
3. `docker-compose up`
4. Go to localhost:8000 and complete the wordpress installation
5. Go to `/wp-admin`
6. Go to `Tools > Site Health` and you should see the first two errors mentioned above
7. (optional) Install the plugin `WP Crontrol`
8. (optional) Go to `Tools > Cron Schedules` then click on the `cron events` tab and you will see the error `There was a problem spawning a call to the WP-Cron system on your site`
9. (optional) You can also click on any of the scheduled cron events and click `run now` and you will get another `curl error 7` error

## Docker Compose File:
```
version: '3'

services:
  # Database
  db:
    image: mysql:5.7
    volumes:
      - db_data:/var/lib/mysql
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: password
      MYSQL_DATABASE: wordpress
      MYSQL_USER: wordpress
      MYSQL_PASSWORD: wordpress
    networks:
      - wpsite
  # phpmyadmin
  phpmyadmin:
    depends_on:
      - db
    image: phpmyadmin/phpmyadmin
    restart: always
    ports:
      - '8080:80'
    environment:
      PMA_HOST: db
      MYSQL_ROOT_PASSWORD: password 
    networks:
      - wpsite
  # Wordpress
  wordpress:
    depends_on:
      - db
    image: wordpress:latest
    ports:
      - '8000:80'
    restart: always
    volumes: ['./:/var/www/html']
    environment:
      WORDPRESS_DB_HOST: db:3306
      WORDPRESS_DB_USER: wordpress
      WORDPRESS_DB_PASSWORD: wordpress
    networks:
      - wpsite
networks:
  wpsite:
volumes:
  db_data:
```

**Comment by wglambert** [comment]
Your port mapping `8000:80` is for connections that go [host]-->[container]. So the container itself is unaware of this port mapping and is listening on `80` while the host is listening on `8000`

**Comment by MinKaizen** [comment]
> Your port mapping `8000:80` is for connections that go [host]-->[container]. So the container itself is unaware of this port mapping and is listening on `80` while the host is listening on `8000`

Ahh I see. Sorry for thinking this was an issue with the image. I'll do some googling and find out how to to set it up properly. If you have any suggestions/pointers, I'd be keen to hear them.

**Comment by btxtiger** [comment]
@MinKaizen What is the solution on this?

**Comment by MinKaizen** [comment]
> @MinKaizen What is the solution on this?

Not sure. I never did end up figuring it out. I ended up using Lando (https://lando.dev)

**Comment by oivinds** [comment]
Why is this closed? Maybe explain in the docs that you will encounter curl error 7 in site health in wp-admin if you use a different port number than 80.

**Comment by btxtiger** [comment]
@oivinds There is a newer issue from me. Seems it's just the test is failing, because they are making an e2e test from the backend side, which will obviously never work with any reverse proxy structure.

**Comment by oivinds** [comment]
> @oivinds There is a newer issue from me. Seems it's just the test that is failing, because they are making an e2e test from the backend side, which will obviously never work with any reverse proxy structure.


My point is that the docker image doc's should be updated, so that people will not waste time on this. So that new people are aware that the Wordpress wp-admin site health feature will not be able to do a loopback request if you use another host port than 80. 
In docker compose:

// no curl 7 error

ports:
      - 80:80

// curl 7 error

ports:
      - 8000:80

**Comment by visionm45** [comment]
I have had the same issue and I was not able to continue until I found a solution. From what I can tell this is only an issue if you are mapping the port to something other than 80.  What I did was edit the apache server config files to match the port that was set for the container.

In /etc/apache2/sites-enabled/000-default.conf
So in my case I added *:8080 to the first line
<VirtualHost *:80 *:8080>

then in /etc/apache2/ports.conf
just below
Listen 80

I added the line
Listen 8080

then after running 
apachectl restart

everything was working as expected
